### ISSUE TITLE
Unable to find GDB

### ISSUE THREAD

**Issue by romangg** [anydistro/bxt#143]
When following [the guide](https://github.com/anydistro/bxt/blob/master/CONTRIBUTING.md#dev-setup) debugging is not possible because it can't find GDB. Running works fine though.

![Screenshot_20240907_170132](https://github.com/user-attachments/assets/f1e2ed33-adb7-4815-9b7d-ad8ea24c56d7)

**Comment by romangg** [comment]
Maybe just needs to be installed to the container?

**Comment by LordTermor** [comment]
It uses lldb

I've [updated](https://github.com/anydistro/bxt/commit/092e4915468db5b1d3cb7b48f5196b3e2ba9ef1f) needed vscode extension, now it uses official `llvm-vs-code-extensions.lldb-dap` instead of third party `vadimcn.vscode-lldb` and should try to run `/bin/lldb-vscode-17`, not `lldb-mi`.

Try to install the extension.

**Comment by romangg** [comment]
I'm pretty sure I installed all extensions it asked for. Let me try again.

**Comment by romangg** [comment]
The extension llvm-vs-code-extensions.lldb-dap is installed in the container. There is `/bin/lldb-vscode-17` but no `/bin/lldb-dab` what the [extension says](https://marketplace.visualstudio.com/items?itemName=llvm-vs-code-extensions.lldb-dap) would be needed.

**Comment by LordTermor** [comment]
They've renamed the executable in llvm 18, we're still 17.

Check `launch.conf` and settings of this extension, I've added [needed path to the config](https://github.com/anydistro/bxt/blob/cc00d84d407982c2bd5073c7ccbc83ba75bc52ef/.vscode/settings.json#L4) but maybe it conflicts with something.


Actually, judging from your screenshot it looks like you're trying to debug using Microsoft's cpptools and not `lldb-dap`, how exactly are you running the debugger?

**Comment by romangg** [comment]
Like that:
![Screenshot_20240908_110234](https://github.com/user-attachments/assets/11d9b0e6-8f3e-4f55-97a7-3821cf249cb5)

This is run form a complete new local checkout of the project. So it should work out-of-the-box without any manual changes necessary. I installed all recommended extensions in the dev container too.

Now that you mention it though, going to the "Run and Debug" screen and pressing there on Start Debugging works though. Doing that once adds another button to the bottom:

![Screenshot_20240908_110830](https://github.com/user-attachments/assets/79314f87-a274-499d-90fc-7b9c17e8fd06)

It would be nice if the "Launch the debugger for the selected target" button is doing something useful too and not error out what is confusing.
